clients/gwtrpc/src/main/java/org/activityinfo/legacy/shared/impl/GetUserProfileFromEmailHandler.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * ActivityInfo
+ * Copyright (C) 2009-2013 UNICEF
+ * Copyright (C) 2014-2018 BeDataDriven Groep B.V.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.activityinfo.legacy.shared.impl;
+
+import com.bedatadriven.rebar.sql.client.*;
+import com.bedatadriven.rebar.sql.client.query.SqlQuery;
+import com.google.gwt.user.client.rpc.AsyncCallback;
+import org.activityinfo.legacy.shared.command.GetUserProfile;
+import org.activityinfo.legacy.shared.command.GetUserProfileFromEmail;
+import org.activityinfo.legacy.shared.model.UserProfileDTO;
+
+public class GetUserProfileFromEmailHandler implements CommandHandlerAsync<GetUserProfileFromEmail, UserProfileDTO> {
+
+    @Override
+    public void execute(final GetUserProfileFromEmail command,
+                        ExecutionContext context,
+                        final AsyncCallback<UserProfileDTO> callback) {
+
+        final String userEmail = command.getUserEmail();
+
+        SqlQuery.select()
+                .appendColumn("u.userId", "id")
```

Comment:
No security? Anyone can execute this request? Why is it necessary?

Reply:
This was because I wanted user to be able to just put in an email and then determine whether that user had an account. 

I think a better way to do this is to only allow the user to select from the list of users already included on the database. I'll scrap this command as a result.